lib/Dialect/Seq/Transforms/LowerSeqToSV.cpp

```diff
@@ -282,29 +283,36 @@ void FirRegLower::lower() {
   if (regs.empty())
     return;
 
-  // Lower the regs to SV regs.
-  SmallVector<RegLowerInfo> toInit;
-  for (auto reg : llvm::make_early_inc_range(regs))
-    toInit.push_back(lower(reg));
+  // Lower the regs to SV regs. Group them by initializer and reset kind.
+  SmallVector<RegLowerInfo> randomInit, presetInit;
+  llvm::MapVector<Value, SmallVector<RegLowerInfo>> asyncResets;
+  for (auto reg : llvm::make_early_inc_range(regs)) {
+    auto svReg = lower(reg);
+    if (!disableRegRandomization) {
+      if (svReg.preset)
+        presetInit.push_back(svReg);
+      else
+        randomInit.push_back(svReg);
+    }
```

Comment:
Should preset initialization be gated under "disabling randomization"?

Reply:
Thanks for catching this, presets should be present when randomization is disabled.